Improve Readme regarding delegation of tasks

in0rdr opened the pull request:
I would like to ensure the mentioned "Go ecosystem" from the Readme on Debian based installations.

For the meta file I was not quite sure if we can include the supported `platforms` only on the collection level and skip the file in the role?

I only tested this on bookworm with a preparation task from my playbook as follows:
```yaml
    - name: Ensure Signalilo build prerequisites (signalilo role)
      ansible.builtin.apt:
        package: '{{ item }}'
        state: present
        update_cache: true
      become: true
      loop:
        - golang
        - make
```

I changed the Ansible module to the more generic `package`, so it might work for other distros in the future as well. I'm not sure about the necessary third party repos that would be required on RHEL based distros though.

Commit:
feat: ensure debian dependencies

Commit:
fix: yaml lint

Commit:
fix: yaml lint

alesz commented on roles/signalilo/tasks/main.yml:
```diff
@@ -1,3 +1,17 @@
+- name: Include os specific vars
```
- this task should be in `source.yml` as it is only needed by source installation method
- task should be delegated to `localhost` as build is not intended to be on target host
- by using `become: true` (which this task requires), playbook executor must have local privileged access. This is not always possible. Suggestion: instead test for presence of `make` and `go`, fail if not available. Installing golang in user space is simple (https://go.dev/doc/install) and allows choice to do it either as user or system-wide.

Commit:
feat: include vars locally

Commit:
feat: undo changes

I undo all the changes, because I was not aware that the compilation
(`source` strategy) also happens locally.

Instead of changing the playbook logic, I propose to improve the Readme.

Commit:
doc: explain task delegation

Commit:
feat: remove uneeded files

ghost commented:
@alesz your comment helped me to understand how the delegation of tasks works in the role. I suggest to keep the playbook logic as is and only improve the Readme a little bit, to make the delegation of the tasks a bit more clear.

Also, we happily use the `source` strategy now, I think a just made a mistake when I tried it the other day.
